### PR TITLE
delete unused functions in pkg/controller directory

### LINKED ISSUE #115088: Cleanup dead code in the codebase
https://github.com/kubernetes/kubernetes/pull/111654 started here but it got big (shocking!) and the contributor got busy.  There are a lot of potential cleanups in there, though, and I hate wasting work.

I see it as an opportunity for someone to rack up some PRs and net-negative LOCs (the best kind!) - just take over that PR, break it up into a bunch of smaller PRs, and send them our way!

A very eager beaver might automate this, but there's a lot of manually-identified low-hanging fruit, I think.

### PR DESCRIPTION
**Pull request by HirazawaUi**
#### What type of PR is this?
/kind cleanup

#### Which issue(s) this PR fixes:

xref #115088

#### Special notes for your reviewer:
This PR follows the content of #111654, the purpose is to split #111654 into many small PRs, this PR only modifies the pkg/controller directory
#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by thockin** [comment]
Thanks!

/lgtm
/approve

File: pkg/controller/lookup_cache.go [hunk 1]
```diff
@@ -52,13 +52,6 @@ type MatchingCache struct {
 	cache *lru.Cache
 }
 
-// NewMatchingCache return a NewMatchingCache, which save label and selector matching relationship.
-func NewMatchingCache(maxCacheEntries int) *MatchingCache {
```
Comment (thockin) [inline pkg/controller/lookup_cache.go:56]: Do we need any of the ` MatchingCache` type?  It seems not, by grepping.  I think you can kill off the type and the methods.

**Comment by HirazawaUi** [comment]
@thockin Excuse me, it appears that there may be some environmental issues causing the e2e tests to fail. Could you please advise me on what steps to take next?

**Review by thockin — COMMENTED** [review]
Thanks!

/lgtm
/approve

**Comment by HirazawaUi** [comment]
> Thanks!
> 
> /lgtm /approve

I apologize for the earlier interruption. I misunderstood the test failure results in the PR (which not only represented failures from the current execution `/retest`),  and I didn't check each test failures carefully enough.

File: pkg/controller/resourcequota/resource_quota_monitor.go [hunk 2]
```diff
@@ -103,19 +103,6 @@ type QuotaMonitor struct {
 	updateFilter UpdateFilter
 }
 
-// NewMonitor creates a new instance of a QuotaMonitor
-func NewMonitor(informersStarted <-chan struct{}, informerFactory informerfactory.InformerFactory, ignoredResources map[schema.GroupResource]struct{}, resyncPeriod controller.ResyncPeriodFunc, replenishmentFunc ReplenishmentFunc, registry quota.Registry) *QuotaMonitor {
```
Comment (soltysh) [inline pkg/controller/resourcequota/resource_quota_monitor.go:107]: @HirazawaUi This function is part of public API and was consumed at least in https://github.com/openshift/cluster-policy-controller/blob/91dfd166cc6a305dfc93b7e45d01fa31e57fd437/pkg/quota/clusterquotareconciliation/reconciliation_controller.go#L104 to my knowledge. I'm not sure if there are other consumers of it or not. But I'd like to ask for a bit more due diligence before removing public function without prior notice. cc @thockin

**Comment by HirazawaUi** [comment]
@soltysh
Please refer to: https://github.com/kubernetes/kubernetes/pull/115103#issuecomment-1503631077
We do not support API compatibility for internal packages.

**Comment by atiratree** [comment]
To fix the regression, I have opened a PR to reintroduce the NewMonitor constructor: https://github.com/kubernetes/kubernetes/pull/120777

**Pull request merged**